src/accessory.ts

```diff
@@ -93,14 +96,32 @@ class PurpleAirSensor implements AccessoryPlugin {
   }
 
   update() {
-    const url = 'https://www.purpleair.com/json?show=' + this.sensor;
+    const url = 'https://www.purpleair.com/json';
+
+    const axiosInstance = axios.create();
+
+    axiosInstance.interceptors.request.use((request) => {
+      // write down your request intercept.
+      return AxiosLogger.requestLogger(request, {
+        logger: this.log
+      });
+    });
 
     if (this.lastReading !== undefined && this.lastReading.updateTimeMs > Date.now() - PurpleAirSensor.MIN_UPDATE_INTERVAL_MS) {
       this.log(`Skipping a fetch because the last update was ${Date.now() - this.lastReading.updateTimeMs} ms ago`);
     } else {
-      this.log(`Fetching ${url}`);
+      this.log(`Fetching`);
+
+      axiosInstance.get(url, {
+        params: {
+          show: this.sensor,
+          key: this.key,
+        },
+      }).then(resp => {
+        if (!resp.data.results[0]) {
+          throw new Error('No sensor found')
```

Comment:
How would the Error propagate? Does it crash the entire plugin if there's a transient server failure?

Reply:
Good question, I thought it would get handled by the catch block but not confident. Refactored to use async/await which has a more clear control flow